src/Microsoft.Windows.CsWin32/templates/PInvokeClassMacros.cs

```diff
@@ -58,4 +58,102 @@ internal class PInvokeClassMacros
+	/// </summary>
+	/// <param name="lParam">The value to be converted.</param>
+	/// <returns>The application command.</returns>
+	/// <remarks>
+	/// Learn more in <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-get_appcommand_lparam">the documentation for this API</see>.
+	/// </remarks>
+	internal static short GET_APPCOMMAND_LPARAM(global::Windows.Win32.Foundation.LPARAM lParam) => unchecked((short)(HIWORD(unchecked((uint)(nint)lParam)) & ~0xF000));
+
+	/// <summary>
+	/// Retrieves the input device type from the specified <see cref="global::Windows.Win32.Foundation.LPARAM"/> value.
+	/// </summary>
+	/// <param name="lParam">The value to be converted.</param>
+	/// <returns>The input device type.</returns>
+	/// <remarks>
+	/// Learn more in <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-get_device_lparam">the documentation for this API</see>.
+	/// </remarks>
+	internal static ushort GET_DEVICE_LPARAM(global::Windows.Win32.Foundation.LPARAM lParam) => unchecked((ushort)(HIWORD(unchecked((uint)(nint)lParam)) & 0xF000));
+
+	/// <summary>
+	/// Retrieves the key state flags from the specified <see cref="global::Windows.Win32.Foundation.LPARAM"/> value.
+	/// </summary>
+	/// <param name="lParam">The value to be converted.</param>
+	/// <returns>The key state flags.</returns>
+	/// <remarks>
+	/// Learn more in <see href="https://learn.microsoft.com/windows/win32/api/winuser/nf-winuser-get_flags_lparam">the documentation for this API</see>.
+	/// </remarks>
+	internal static ushort GET_FLAGS_LPARAM(global::Windows.Win32.Foundation.LPARAM lParam) => LOWORD(unchecked((uint)(nint)lParam));
+
+	/// <summary>
+	/// Retrieves the key state from the specified <see cref="global::Windows.Win32.Foundation.LPARAM"/> value.
+	/// </summary>
+	/// <param name="lParam">The value to be converted.</param>
+	/// <returns>The key state.</returns>
+	/// <remarks>
+	/// Learn more in <see href="https://learn.microsoft.com/windows/win32/inputdev/wm-appcommand">the documentation for this API</see>.
+	/// </remarks>
+	internal static ushort GET_KEYSTATE_LPARAM(global::Windows.Win32.Foundation.LPARAM lParam) => LOWORD(unchecked((uint)(nint)lParam));
+
+	/// <summary>
+	/// Retrieves the key state from the specified <see cref="global::Windows.Win32.Foundation.WPARAM"/> value.
+	/// </summary>
+	/// <param name="wParam">The value to be converted.</param>
+	/// <returns>The key state.</returns>
+	internal static ushort GET_KEYSTATE_WPARAM(global::Windows.Win32.Foundation.WPARAM wParam) => LOWORD(unchecked((uint)(nuint)wParam));
+
+	/// <summary>
+	/// Retrieves the hit-test value from the specified <see cref="global::Windows.Win32.Foundation.WPARAM"/> value.
+	/// </summary>
+	/// <param name="wParam">The value to be converted.</param>
+	/// <returns>The hit-test value.</returns>
+	internal static uint GET_NCHITTEST_WPARAM(global::Windows.Win32.Foundation.WPARAM wParam) => unchecked((uint)(nuint)wParam);
```

Comment:
Is this right?  This is documented as `The return value is the low-order word representing the hit-test value`.

https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-get_nchittest_wparam